app/Http/Middleware/HasBankAccount.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+
+use \App\Tenant as Tenant;
+
+class HasBankAccount
+{
+		/**
+		 * Handle an incoming request.
+		 *
+		 * @param  \Illuminate\Http\Request  $request
+		 * @param  \Closure  $next
+		 * @return mixed
+		 */
+		public function handle($request, Closure $next)
+		{
+			$tenant = new Tenant();
```

Comment:
here we define a new tenant which isnt what we want. we should be retrieving the tenant from the logged in user like so 

https://github.com/purplenimbus/edu-api/blob/master/app/Http/Controllers/CourseController.php#L35

```suggestion
			   $user = Auth::user();
                           $tenant = $user->tenant()->first();
```

this should work since we already have access to the logged in user at this point

Reply:
This was the place I was experiencing some difficulty. I was trying to access the tenant model using the user_id in the request data.